torchgeo/datasets/l7irish.py

```diff
@@ -0,0 +1,293 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""L7 Irish dataset."""
+
+import glob
+import os
+import re
+from typing import Any, Callable, Dict, List, Optional, Sequence, cast
+
+import matplotlib.pyplot as plt
+import torch
+from rasterio.crs import CRS
+from torch import Tensor
+
+from .geo import RasterDataset
+from .utils import BoundingBox, download_url, extract_archive
+
+
+class L7Irish(RasterDataset):
+    """L7 Irish dataset.
+
+    The `L7 Irish <https://www.sciencebase.gov/catalog/item/573ccf18e4b0dae0d5e4b109>`__ dataset
+    is based on Landsat 7 Enhanced Thematic Mapper Plus (ETM+) Level-1G scenes.
+    Manually generated cloud masks are used to train and validate cloud cover assessment algorithms,
+    which in turn are intended to compute the percentage of cloud cover in each scene.
```

Comment:
Some of these lines are over 90 characters. Can you wrap these? I know we ignored line length because of the URL but we shouldn't exceed line length when we don't have to.